presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSource.java

```diff
@@ -363,6 +364,17 @@ public ConnectorPageSource getPageSource()
         if (fromType instanceof VarcharType && (toHiveType.equals(HIVE_BYTE) || toHiveType.equals(HIVE_SHORT) || toHiveType.equals(HIVE_INT) || toHiveType.equals(HIVE_LONG))) {
             return new VarcharToIntegerNumberCoercer<>((VarcharType) fromType, toType);
         }
+        if (fromType instanceof VarcharType && toType instanceof VarcharType) {
+            VarcharType toVarcharType = (VarcharType) toType;
+            VarcharType fromVarcharType = (VarcharType) fromType;
+
+            if (toVarcharType.narrowerThan(fromVarcharType)) {
+                return new VarcharCoercer(fromVarcharType, toVarcharType);
+            }
+
+            // Coercion not worth the copying penalty when toType is wider
+            return null;
```

Comment:
Return `Functions.identity()` instead for now.

As a followup, yes it makes sense to have `createCoercer` return Optional.
Then the `fromType.equals(toType)` check should be moved inside the method.